sdk/core/azure-json/src/main/java/com/azure/json/JsonObject.java

```diff
@@ -0,0 +1,494 @@
+     *
+     * @param key specifying the key to match a property with to find the
+     * respective value pair of
+     * @return JsonElement object representing the respective value pair of the
+     * search key
+     */
+    public JsonElement getValueByKey(String key) {
+        return properties.get(key);
+    }
+
+//    /**
+//     * Returns the String representation of the JsonObject object
+//     *
+//     * @return String representation of the JsonObject object
+//     */
+//    public String toJson() {
+//        // String reference that will store the resulting JSON string output to
+//        // be returned by toJson
+//        String jsonOutput = "";
+//
+//        // Iterating over the keys in the key set of the properties hash map,
+//        // ....
+//        for(Iterator<String> itr = properties.keySet().iterator(); itr.hasNext();) {
+//            // Get the next key String from the properties hash map
+//            String key = itr.next();
+//
+//            // Including the double quotations marks, via escape characters, which
+//            // will surround the given key in the String representation of the
+//            // JSON object. All keys must be in quotes.
+//            jsonOutput += "\""+ key + "\": ";
+//
+//            // Case: the value pair is another JsonObject, therefore the nested
+//            // JSON object can be stringified via recursively calling toJson and
+//            // then appending the returned String to jsonoutput
+//            if(this.getProperty(key) instanceof JsonObject) {
+//                jsonOutput += ((JsonObject) this.getProperty(key)).toJson();
+//            }
+//            // Case: the value pair is a JsonArray, therefore the whole JSON
+//            // arary can be stringified via its toJson method, and the returned
+//            // String can be appended to jsonOutput
+//            else if(this.getProperty(key) instanceof JsonArray) {
+//                jsonOutput += ((JsonArray) this.getProperty(key)).toJson();
+//            }
+//            // Case: the value pair is not a JsonObject or JsonArray, therefore
+//            // it can be simply converted to a String object and appended to
+//            // jsonOutput
+//            else { jsonOutput += this.getProperty(key); }
+//
+//            // Case: haven't reached the end of the JSON Object, therefore there
+//            // are more properties that must be converted to String objects and
+//            // appended to the jsonOutput.
+//            // Properties are separated by commas.
+//            if(itr.hasNext()) { jsonOutput += ", "; }
+//        }
+//
+//        // Returning the resulting String representation of the JsonArray object
+//        // JSON objects are delimited by opening and closing curly brackets.
+//        return "{" + jsonOutput + "}";
+//    }
+
+    public JsonObject setProperty(String key, JsonElement element) throws IllegalArgumentException {
+        properties.put(key, element);
```

Comment:
When is the `IllegalArgumentException` thrown? Is it thrown if the element is null or key already exists?